src/GeniusYield/TxBuilder/Class.hs

```diff
@@ -520,6 +546,14 @@ utxoDatum utxo = case utxoOutDatum utxo of
         Nothing -> pure . Left $ GYInvalidDatum x
         Just a  -> pure $ Right (utxoAddress utxo, utxoValue utxo, a)
 
+-- | Obtain original datum representation of an UTxO.
+utxoDatumPureHushed :: Plutus.FromData a => (GYUTxO, Maybe GYDatum) -> Maybe (GYTxOutRef, (GYAddress, GYValue, a))
+utxoDatumPureHushed (_utxo, Nothing) = Nothing
+utxoDatumPureHushed (GYUTxO {..}, Just d) =
+  case Plutus.fromBuiltinData $ datumToPlutus' d of
+    Nothing -> Nothing
+    Just d' -> Just (utxoRef, (utxoAddress, utxoValue, d'))
```

Comment:
Maybe we can replace the case with something like:

import           Control.Lens ((<&>))
utxoDatumPureHushed (GYUTxO {..}, Just d) = Plutus.fromBuiltinData (datumToPlutus' d)
                                                                               <&> \d' -> (utxoRef, (utxoAddress, utxoValue, d'))

Reply:
Yes, as `Nothing` case isn't having any special treatment, we can use functorial context here.